rfd/0083-prevent-user-role-self-lock.md

```diff
@@ -0,0 +1,108 @@
+| role edit              | _admin_                                           | _admin_ edits own role to remove user/role editing capabilities                        |
+| SSO connector removal  | _SSO admin_                                       | _SSO admin_ removes SSO connector. _SSO admin_ logs out.                               |
+| SSO user/group removal | _SSO admin_                                       | Someone on the SSO side removes _SSO admin_ user or team/group containing _SSO admins_ |
+
+Teleport already protects against a similar case in the following scenario:
+
+| action       | initial state | change                           |
+| ------------ | ------------- | -------------------------------- |
+| role removal | _admin_       | _admin_ tries to remove own role |
+
+The error message is displayed: _failed to delete role that still in use by a user. Check system server logs for more details_.
+
+There is one more case when a user can lock himself out:
+
+| initial state | change                                                                                    |
+| ------------- | ----------------------------------------------------------------------------------------- |
+| _admin_       | _admin_ loses acess to own account (eg. he stores credentials and recovery codes locally) |
+
+However, requiring a second admin to guard this situation is not good for user experience. Existing and new clusters will have to add a second _admin_.
+
+## Terminology
+
+- _admin_: local user with user/role editing capabilities
+- _SSO admin_: user with user/role editing capabilities added via SSO connector (eg. GitHub)
+
+## Why
+
+When a user locks themselves out, they are unable to further manage the teleport cluster. We want to prevent that since this is bad for the user experience and prevents them from using the product.
+
+Further, for our Cloud offering, we don't perform account resets as a policy. The only recourse for a customer is to delete their cluster & deploy a new one. This inevitably leads to data loss.
+
+On-prem customers could work around this "locked out" state by running `tctl` locally on the auth server to create a new admin user.
+
+## Details
+
+> Context: SSO users
+>
+> Teleport has this feature called SSO auth. It allows for the addition of an SSO connector to a cluster and maps SSO users/user groups to teleport roles. Currently, it works this way:
+>
+> 1. Authorized teleport users can add SSO connector (eg. GitHub SSO https://goteleport.com/docs/setup/admin/github-sso)
+> 2. GitHub users that are members of a particular organization and the team can log in as teleport users with a defined role.
+> 3. Users will show up in the users' list in UI but they will disappear when their cert expires (even when we remove the SSO connector).
+> 4. SSO users can have full access.
+
+The solution here could be to introduce a new rule:
+
+- There should be at least _admin_
```

Comment:
Do you mean that "There should be at least one _admin_ user" ? The sentence seems incomplete.

Reply:
This describes a state where some user is able to manage cluster